backend/server/listen.py

```diff
@@ -1,312 +1,21 @@
-from server.modeltypes import DataModel, HandleFolder, HandleFile, FileWrite
-from fastapi import FastAPI, HTTPException, WebSocket, Security, Query
-from fastapi.security import APIKeyHeader
+from .routers import api, ws, api_key
+from fastapi import FastAPI
 from fastapi.middleware.cors import CORSMiddleware
-from server.session.manager import Manager
-import shutil
-import json 
-import os
 
 app = FastAPI()
 
-KEY_MAP = {"OpenAI": "OPENAI_API_KEY", "Anthropic": "ANTHROPIC_API_KEY"}
+#Include routers
+app.include_router(api_key.router, prefix="/api-key", tags=["api-key"])
+app.include_router(api.router, prefix="/api", tags=["api"])
+app.include_router(ws.router, prefix="/ws", tags=["ws"])
```

Comment:
I tried to test this path locally. I got the following errors

`WebSocket connection to 'ws://localhost:3001/ws' failed: `

```
backend-1   | INFO:     ('192.168.65.1', 30997) - "WebSocket /ws" 403
backend-1   | INFO:     connection rejected (403 Forbidden)
```

The connection is made inside `frontend/src/services/session.ts`

Has this worked locally for you? One way to test this would be to run an Agent recipe.